### ISSUE TITLE
Error setting TTL index on collection 

### ISSUE THREAD

**Issue by riston** [kcbanner/connect-mongo#80]
I am using mongolab for hosting the sessions and the app is under the heroku host, some times the starting application gives the error. 

```
**/node_modules/connect-mongo/node_modules/mongodb/lib/mongodb/connection/server.js:529
        throw err;
              ^
Error: Error setting TTL index on collection : sessions <MongoError: not authorized for query on herouku*****.system.indexes>
```

After going further it seems there is problem with executing the ensureIndex, does this get executed evert connect? 

Or can this just be removed and be set from mongo shell ?

**Comment by techjeffharris** [comment]
I have been really busy with my day-job, sorry!  

@malixsys, I have been under the impression that MongoStore only needs a callback if the mongo/mongoose connection has not initialized.   You're saying that I need to use a callback for MongoStore even if my db connection is open?

@jdesboeufs, if I upgrade to connect-mongo:master, this issue should theoretically be resolved? A lot has changed, so I want to get up-to-date instructions before I check resolution :)

Thanks for all your hard work!

**Comment by malixsys** [comment]
@techjeffharris I use a callback to wait for the DB connection before starting my server.listen() so that the store does not try to create indexes too soon...

**Comment by jdesboeufs** [comment]
@malixsys Using a callback must now be avoided since `connect-mongo` will wait for the database connection.

@techjeffharris You can upgrade to 0.5.3, mongoose support has been improved and I need feedback :)

**Comment by techjeffharris** [comment]
@jdesboeufs Sorry, I have been busy with a side project.. I'll cross my fingers and open a new issue if I run into problems :)  Thanks again!
